lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -135,3 +137,21 @@ defaultRetryPolicy =
     limitRetriesByCumulativeDelay (60 * second) (constantDelay (1 * second))
   where
     second = 1000*1000
+
+-- | The 'nextBlocks' function should never return duplicate slots, blocks out
+-- of order, or blocks before the requested starting point. Doing so would break
+-- wallet invariants. This validation function should be used by 'NetworkLayer'
+-- implementations to ensure proper error handling.
+validateNextBlocks
+    :: Functor m
+    => BlockHeader
+    -> ExceptT ErrGetBlock m [Block tx]
+    -> ExceptT ErrGetBlock m [Block tx]
+validateNextBlocks start = mapExceptT (fmap checkValid)
+  where
+    checkValid (Right bs)
+        | areValid (map (slotId . header) bs) = Right bs
+        | otherwise = Left ErrGetBlocksOutOfOrder
+    checkValid (Left e) = Left e
+    start' = slotId start
+    areValid ss = and $ zipWith (\a b -> a >= start' && b > a) ss (drop 1 ss)
```

Comment:
I suspect this should be `a > start'` rather than `a >= start'`. At least, that's the way the `jormungandr` and `http-bridge` implementations appear to work (as I have understood them).

Reply:
I would suggest writing it in this order:
```hs
start < a && a < b
```
It's easier to see the equivalence with `start < a < b` at a glance.